src/lib/operators/abstract_aggregate_operator.hpp

```diff
@@ -90,19 +93,73 @@ class AggregateFunctionBuilder<ColumnDataType, AggregateType, AggregateFunction:
   }
 };
 
+template <typename ColumnDataType, typename AggregateType>
+class AggregateFunctionBuilder<ColumnDataType, AggregateType, AggregateFunction::StdDevSamp> {
+ public:
+  auto get_aggregate_function() {
+    return [](const ColumnDataType& new_value, std::optional<AggregateType>& current_primary_aggregate,
+              std::vector<AggregateType>& current_secondary_aggregates) {
+      if constexpr (std::is_arithmetic_v<ColumnDataType>) {
+        // Welford's online algorithm
+        // https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance#Welford's_online_algorithm
+        // For a new value, compute the new count, new mean and the new squared_distance_from_mean.
+        // mean accumulates the mean of the entire dataset.
+        // squared_distance_from_mean aggregates the squared distance from the mean.
+        // count aggregates the number of samples seen so far.
+        // The sample standard deviation is stored as current_primary_aggregate.
+        if (current_secondary_aggregates.empty()) {
+          current_secondary_aggregates.resize(3);
+          // current_secondary_aggregates[0]:   storage for count
+          // current_secondary_aggregates[1]:   storage for mean
+          // current_secondary_aggregates[2]:   storage for squared_distance_from_mean
+        }
+
+        // get values
+        double count = current_secondary_aggregates[0];
+        double mean = current_secondary_aggregates[1];
```

Comment:
how about making this a reference so you don't need the "store values" step below.